File: neet-code-150/arrays/two_sum.cpp
#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;

vector<int> twoSum(vector<int> &nums, int target) {
  unordered_map<int, int> seen;

  for (int i = 0; i < int(nums.size()); i++) {
    int diff = target - nums[i];
    if (seen.count(diff) > 0) {
      return {seen[diff], i};
    }

    seen[nums[i]] = i;
  }

  return {};
}

int main() {
  vector<int> nums = {3, 2, 4};

  vector<int> result = twoSum(nums, 6);

  if (result.size() == 0) {
    cout << "Result is empty" << endl;
  }

  for (int i = 0; i < int(result.size()); i++) {
    cout << result[i] << endl;
  }

  return 0;
}

File: neet-code-150/arrays/two_sum.py
def twoSum(nums: list[int], target: int) -> list[int]:
    seen = {}
    res = []

    for i in range(len(nums)):
        diff = target - nums[i]

        if diff in seen:
            res = [seen[diff], i]

        seen[nums[i]] = i

    return res


print(twoSum([2, 7, 11, 15], 9))
